examples/extensions/commands/run_example.py

```diff
@@ -0,0 +1,78 @@
+import os
+import shutil
+from contextlib import contextmanager
+
+from test.examples_tools import run, tmp_dir
+
+CONAN_CACHE_FOLDER = run("conan config home").split(":", maxsplit=1)[1].strip()
+EXTENSIONS = os.path.join(CONAN_CACHE_FOLDER, "extensions")
+CUSTOM_COMMAND_FILES = [
+    os.path.join(EXTENSIONS, "commands", "cmd_clean.py")
+]
+
+
+def copy_commands_folder():
+    clean_command = os.path.join(os.path.dirname(os.path.realpath(__file__)), "cmd_clean.py")
+    conan_commands_folder = os.path.join(EXTENSIONS, "commands")
+    os.makedirs(conan_commands_folder, exist_ok=True)
+    shutil.copy(clean_command, conan_commands_folder)
+    for f in CUSTOM_COMMAND_FILES:
+        assert os.path.exists(f)
+
+
+def remove_custom_commands():
+    for f in CUSTOM_COMMAND_FILES:
+        os.remove(f)
+
+
+@contextmanager
+def custom_commands():
+    try:
+        copy_commands_folder()
+        yield
+    finally:
+        remove_custom_commands()
+
+
+with custom_commands():
+    # 1. Check the custom command is appearing in conan help
+    output = run("conan -h")
+    assert "Custom commands\nclean" in output.replace("\r\n", "\n")
+
+    # 2. Create several packages
+    with tmp_dir("clean_hello"):
+        run("conan new -d name=clean_hello -d version=1.0 cmake_lib")
+        # Non-deterministic library (changing PREV each time)
+        with open(os.path.join("src", "clean_hello.cpp"), "w") as f:
```

Comment:
No need to compile things, just create different recipe revisions, then 1 empty package per revision. Much faster, tests the same.